### PR TITLE
[@mantine/form]: valibot mantine form resolver

### PR DESCRIPTION
**Pull request by annibuliful**
Add [valibot](https://valibot.dev/) to mantine `useForm`

### CODE REVIEW

File: src/mantine-demos/src/demos/form/Form.demo.valibot.tsx [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  username: string('Username is required', [
+    minLength(8, 'Username must have at least 8 characters'),
+  ]),
+  password: string('Password is required', [
+    minLength(8, 'Password must have at least 8 characters'),
+  ]),
+});
+
+function Demo() {
+  const form = useForm({
```
Comment (Songkeys) [inline src/mantine-demos/src/demos/form/Form.demo.valibot.tsx:20]: In typescript, we might need the generic `<FormData>` type. To type it in valibot:

```ts
import { type Input } from 'valibot';

type FormData = Input<typeof schema>;
```

Comment (Songkeys) [reply src/mantine-demos/src/demos/form/Form.demo.valibot.tsx:22]: Ah, I just revisited the [docs](https://mantine.dev/form/validation/#zod), it doesn't specify the generic typing usage though it's a `.tsx` file. I also forgot where I saw that usage. I think we can drop it then.

Comment (annibuliful) [reply src/mantine-demos/src/demos/form/Form.demo.valibot.tsx:20]: Sure, removed it

**Review by Songkeys — APPROVED** [review]
LGTM. Also tested on my end. /cc @rtivital

**Comment by annibuliful** [comment]
I'm happy to do thai

**Pull request closed without merging**